jerry-core/ecma/builtin-objects/ecma-builtin-global.cpp

```diff
@@ -647,23 +652,23 @@ ecma_builtin_global_object_decode_uri_helper (ecma_value_t uri __attr_unused___,
       uint32_t min;
       uint32_t character;
 
-      if (decoded_byte < 0xe0)
+      if (decoded_byte < LIT_UTF8_2_BYTE_MASK)
       {
         count = 1;
-        min = 0x80;
-        character = decoded_byte & 0x1f;
+        min = LIT_UTF8_2_BYTE_CODE_POINT_MIN;
+        character = decoded_byte & LIT_UTF8_LAST_5_BITS_MASK;
       }
-      else if (decoded_byte < 0xf0)
+      else if (decoded_byte < LIT_UTF8_3_BYTE_MASK)
       {
         count = 2;
-        min = 0x800;
-        character = decoded_byte & 0x0f;
+        min = LIT_UTF8_3_BYTE_CODE_POINT_MIN;
+        character = decoded_byte & LIT_UTF8_LAST_4_BITS_MASK;
       }
       else
       {
         count = 3;
-        min = 0x1000;
-        character = decoded_byte & 0x07;
+        min = LIT_UTF8_4_BYTE_CODE_POINT_MIN;
+        character = decoded_byte & LIT_UTF8_LAST_3_BITS_MASK;
       }
```

Comment:
Lines 655-672 are similar to  lines 48-65 of lit-strings.cpp (`lit_is_utf8_string_valid` function), possibly we could extract them into function like
`lit_utf8_size_t lit_utf8_analyze_first_byte (lit_utf8_byte_t first_byte, lit_code_point_t *min, lit_code_point_t *character)`